ministryofjustice/analytics-platform-control-panel
# Feature refactor model service

## What

Goal of this PR was to extract some of the ORM code from the service layer to reduce coupling.

Views are clean and call a method on model.

Model is clean and calls `service` functions passing its instance attributes as params.  Service does not need to know about django model.  Service function could be called by a command by just passing params.

Service layer does a bit of orchestrating and calls lower level aws wrapper functions.

Each layer is tested with each other.

## Issues

- Naming of model methods.
- Refactor out policy name and arn etc.
- Now we have django models with methods hitting aws, is this ok? Do we want to further pull out this to different classes to clean up and make clearer?

## How to review

1. Check design.
2. Run tests.
